### PR TITLE
Add shelved date, populate from Goodreads import

### PR DESCRIPTION
**Pull request by cincodenada**
`shelved_date` defaults to current datetime for new books, and uses the "Date Added" date from Goodreads imports. The shelf page now uses this field for its Shelved Date column, and shelfbooks are sorted by it first.

This may be part of a larger question, but I'm torn on what to do about timezones - I considered making shelved_date a DateField instead of DateTimeField, which I think it should be properly, as that's all we have on imports, but then when folks add a book in Bookwyrm, we'd tag it with whatever date it happened to be in UTC, which isn't necessarily what users would want.

That said, I'm not sure how we're displaying dates, so we may display that date anyway - but not storing the datetime a book was added makes it so we can't recover the "correct" date even if we do (at any point) have user timezone preferences.

### CODE REVIEW

File: bookwyrm/tests/importers/test_goodreads_import.py [hunk 1]
```diff
@@ -13,6 +15,10 @@
 from bookwyrm.settings import DOMAIN
 
 
+def make_date(*args):
+    return datetime.datetime(*args, tzinfo=pytz.UTC)
```
Comment (cincodenada) [inline bookwyrm/tests/importers/test_goodreads_import.py:19]: I don't remember if we like free functions like this, but it seemed just laborious enough to use one to me. I'm not wedded to it tho.

File: bookwyrm/tests/importers/test_goodreads_import.py [hunk 2]
```diff
@@ -130,22 +136,25 @@ def test_handle_imported_book(self):
...
         self.assertEqual(readthrough.book, self.book)
-        # I can't remember how to create dates and I don't want to look it up.
-        self.assertEqual(readthrough.start_date.year, 2020)
-        self.assertEqual(readthrough.start_date.month, 10)
-        self.assertEqual(readthrough.start_date.day, 21)
-        self.assertEqual(readthrough.finish_date.year, 2020)
-        self.assertEqual(readthrough.finish_date.month, 10)
-        self.assertEqual(readthrough.finish_date.day, 25)
+        self.assertEqual(readthrough.start_date, make_date(2020, 10, 21))
+        self.assertEqual(readthrough.finish_date, make_date(2020, 10, 25))
```
Comment (cincodenada) [inline bookwyrm/tests/importers/test_goodreads_import.py:146]: I was feeling ambitious and looked it up :smile:

File: bookwyrm/views/shelf.py [hunk 3]
```diff
@@ -69,7 +69,8 @@ def get(self, request, username, shelf_identifier=None):
             reviews = privacy_filter(request.user, reviews)
 
         books = books.annotate(
-            rating=Subquery(reviews.values("rating")[:1])
+            rating=Subquery(reviews.values("rating")[:1]),
+            shelved_date=F("shelfbook__shelved_date"),
```
Comment (cincodenada) [inline bookwyrm/views/shelf.py:73]: I'm not sure if is the best way to get at the `shelved_date`, I tried accessing it via `book.shelfbook__shelved_date` in the template to no success.

**Comment by cincodenada** [comment]
I think this should resolve #1174

File: bookwyrm/tests/importers/test_goodreads_import.py (hunk 1 above)
Comment (mouse-reeve) [reply bookwyrm/tests/importers/test_goodreads_import.py:19]: I haven't formulated a general opinion on it (seems fine?) but for tests I see absolutely no problem whatsoever. It's handy!

File: bookwyrm/tests/importers/test_goodreads_import.py (hunk 2 above)
Comment (mouse-reeve) [reply bookwyrm/tests/importers/test_goodreads_import.py:146]: Yessssss thank you for being less lazy than me

File: bookwyrm/views/shelf.py (hunk 3 above)
Comment (mouse-reeve) [reply bookwyrm/views/shelf.py:73]: I think this is good, I don't think the templates can do that kind of advanced lookup, and it's cleaner to do it in the view anyway, imo

**Pull request merged**